### PR TITLE
deposit: fix deleting function

### PR DESCRIPTION
**Pull request by hachreak**
* Fixes the delete function to support cascading deletion on PostgreSQL.
  (addresses #3218)

* PEP8/257 code style improvements.

Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

File: invenio/modules/deposit/testsuite/test_deposit_models.py [hunk 1]
```diff
@@ -57,9 +63,13 @@ def test_create(self):
...
         assert Deposition.get(d.id).type == self.DefaultType
-        d = Deposition.create(user, type=self.AnotherType)
-        assert d.type == self.AnotherType
-        assert Deposition.get(d.id).type == self.AnotherType
+        d2 = Deposition.create(user, type=self.AnotherType)
+        assert d2.type == self.AnotherType
+        assert Deposition.get(d2.id).type == self.AnotherType
+
+        # remove the records
+        Deposition.delete(d)
```
Comment (lnielsen) [inline invenio/modules/deposit/testsuite/test_deposit_models.py]: Could you please add a test case that checks if both all the underlying BibWorkflowObjects (multiple) and the associated Workflow object has been removed and nothing more than that.

Comment (hachreak) [inline invenio/modules/deposit/testsuite/test_deposit_models.py]: On which objects I have to test the existence/not existence?

Comment (lnielsen) [inline invenio/modules/deposit/testsuite/test_deposit_models.py]: Deposition is just a wrapper a around a ``BibWorkflowObject`` (workflows module) so you should test something like:

```python
bwl_d2_count = BibWorkflowObject.query.filter_by(uuid=d2.workflow_object.id_workflow).count()
d.delete()
assert Workflow.query.filter_by(uuid=d.workflow_object.id_workflow).count() == 0
assert  BibWorkflowObject.query.filter_by(id_workflow=d.workflow_object.id_workflow).count() == 0
```
and that no other objects where deleted, e.g.:

```python
assert Workflow.query.filter_by(uuid=d2.workflow_object.id_workflow).count() == 1
assert BibWorkflowObject.query.filter_by(uuid=d2.workflow_object.id_workflow).count() == bwl_d2_count
#...
d2.delete()
```

Comment (hachreak) [inline invenio/modules/deposit/testsuite/test_deposit_models.py]: something like this? 

https://github.com/inveniosoftware/invenio/blob/master/invenio/modules/workflows/testsuite/test_workflows_model.py#L28

Comment (lnielsen) [inline invenio/modules/deposit/testsuite/test_deposit_models.py]: Similar except that above tests Deposit/Workflow integration, and in the example above you only have one BibWorkflowObject, whereas you can normally have many.

It's from experience that this is a soft spot between the two modules.

**Pull request merged**